linode_metadata/metadata_client.py

```diff
@@ -300,8 +315,34 @@ def _api_call(
         url = self._prepare_url(endpoint)
         request_params = self._get_api_call_params(url, headers, method, body)
 
+        if self._debug:
+            if self._debug_file is None:
+                self._print_request_debug_info(request_params, sys.stderr)
+            else:
+                if os.path.exists(self._debug_file):
+                    with open(self._debug_file, "a", encoding="UTF-8") as file:
+                        self._print_request_debug_info(request_params, file)
+                else:
+                    print(
+                        "No debug file exists to write. "
+                        + "Please check the file path or use default output."
+                    )
+
         response: Response = method_func(**request_params)
 
+        if self._debug:
+            if self._debug_file is None:
+                self._print_response_debug_info(response, sys.stderr)
+            else:
+                if os.path.exists(self._debug_file):
+                    with open(self._debug_file, "a", encoding="UTF-8") as file:
+                        self._print_response_debug_info(response, file)
+                else:
+                    print(
+                        "No debug file exists to write. "
+                        + "Please check the file path or use default output."
```

Comment:
```suggestion
                        "No debug file exists to write. "
                        "Please check the file path or use default output."
```
`+` is not necessary in this because Python supports connected strings concatenation.